### PR TITLE
SpeechResModel implemented with tfjs

### PR DESCRIPTION
**Pull request by ljj7975**
< Summary >
- SpeechResModel implemented

< Things to Note >
- for `tf.layers.batchNormalization`, Affine = False option on `torch.nn.BatchNorm2d` is equal to gamma = 1 and beta = 0 (https://discuss.pytorch.org/t/affine-parameter-in-batchnorm/6005)
- last two layers
```
x = x.view(x.size(0), x.size(1), -1) # shape: (batch, feats, o3)
x = torch.mean(x, 2)
```
is equal to `tf.layers.averagePooling2d`

### CODE REVIEW

File: SpeechModel.js [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
 
 	constructor() {
-		speechModel = this;
-		// for training
-		this.iteration = 5;
-
 		// this config is place holder (copied from CNN_TRAD_POOL2)
 		this.config = {
 			input_shape : [40, 100, 1],
+			n_labels : 10,
```
Comment (daemon) [inline SpeechModel.js]: IIRC I think `n_labels` should be 12, since we have `unknown` and `silence` on top of the existing ten keywords -- did you account for those additional two elsewhere in the code?

File: SpeechResModel.js [hunk 2]
```diff
@@ -1,41 +1,200 @@
...
-			n_labels : 12,
+			input_shape : [40, 100, 1],
+			n_labels : 10,
 			n_layers : 6,
 			n_feature_maps : 45,
-			res_pool : (4, 3),
+			conv_size : [3, 3],
+			res_pool : [3, 4],
+			conv_stride : [1, 1],
 			use_dilation : false
```
Comment (daemon) [inline SpeechResModel.js]: Should `dilation` be on or off by default?

File: SpeechResModel.js [hunk 3]
```diff
@@ -1,41 +1,200 @@
...
+			})
+		}
+
+		// dilation = config["use_dilation"]
+        // if dilation:
+		if (this.config['use_dilation']) {
+			// 	self.convs = [nn.Conv2d(n_maps, n_maps, (3, 3), padding=int(2**(i // 3)), dilation=int(2**(i // 3)), 
+            //     bias=False) for i in range(n_layers)]
+			// => torch.nn.Conv2d(in_channels, out_channels, kernel_size, stride=1, padding=0, dilation=1, groups=1, bias=True)
+			//     self.add_module("conv{}".format(i + 1), conv)
```
Comment (daemon) [inline SpeechResModel.js]: Should we clean up some of this commented-out code?

File: SpeechResModel.js [hunk 4]
```diff
@@ -1,41 +1,200 @@
...
 	// Our actual model
 	compile() {
-		this.model = tf.sequential();
+		// input layer
+		const input = tf.input({shape: this.config['input_shape']});
+		let x = input; // [40, 100, 1]
+		let y, old_x;
+
+		// for i in range(self.n_layers + 1):
+		for (var i  = 0; i < 3; i++) {
```
Comment (daemon) [inline SpeechResModel.js]: Shouldn't we use `n_layers` here instead?

File: SpeechModel.js (hunk 1 above)
Comment (ljj7975) [reply SpeechModel.js]: oh this config is temporary as mentioned in the comment. Eventually we want this config passed in as part of constructor argument

Comment (daemon) [reply SpeechModel.js]: Cool.
